src/main/scala/org/clulab/beliefs/BeliefProcessor.scala

```diff
@@ -78,6 +86,27 @@ class BeliefProcessor(val processor: Processor,
 
     nounCount > 1 || (nounCount > 0 && verbCount > 0)
   }
+
+  private def isPropositionWithTheme(belief: Mention, beliefTheme: Mention): Boolean = {
+    // accounts for examples like <Belief theme> is believed to <belief>, e.g., `This politician is believed to be a narcissist.`
+    val sent = belief.sentenceObj
+    val tags = sent.tags.get
+    val beliefSpan = belief.tokenInterval
+    val beliefThemeSpan = beliefTheme.tokenInterval
+    var nounCount = 0
+    var verbCount = 0
+    for(i <- beliefSpan.start until beliefSpan.end) {
+      if(tags(i).startsWith("NN")) nounCount += 1
+      else if(tags(i).startsWith("VB")) verbCount += 1
+    }
```

Comment:
Maybe

```scala
val nounCount = tags.view(beliefSpan.start, beliefSpan.end).count(_.startsWith("NN")) + ...
```

or a mini function that takes a tokenInterval and a prefix since there are 4 times that it happens.

```scala
def countTags(tokenInterval: TokenInterval, prefix: String): Int =
    tags.view(tokenInterval.start, tokenInterval.end).count(_.startsWith(prefix))

val nounCount = countTags(belief.tokenInterval, "NN") + countTags(beliefThemeSpan, "NN")
val verbCount = ...
```
Maybe it should take two intervals so that "NN" isn't repeated.